mscore/pianoview.cpp

```diff
@@ -499,10 +502,110 @@ int PianoView::pixelXToTick(int pixX) {
 //   tickToPixelX
 //---------------------------------------------------------
 
-int PianoView::tickToPixelX(int tick) {
+int PianoView::tickToPixelX(int tick)
+      {
       return static_cast<int>(tick + MAP_OFFSET) * _xZoom;
       }
 
+
+//---------------------------------------------------------
+//   event
+//---------------------------------------------------------
+
+bool PianoView::event(QEvent *event)
+    {
+        if (event->type() == QEvent::KeyPress) {
+              QKeyEvent *ke = static_cast<QKeyEvent *>(event);
+
+              bool success = handleKeyPressEvent(ke);
+              if (success)
+                  return true;
+              }
+
+        return QWidget::event(event);
+    }
+
+
+//---------------------------------------------------------
+//   keyPressEvent
+//---------------------------------------------------------
+
+bool PianoView::handleKeyPressEvent(QKeyEvent *event)
+    {
+    bool ctrl = event->modifiers() & Qt::ControlModifier;
+    bool alt = event->modifiers() & Qt::AltModifier;
+
+      qDebug("Keypress: %d", event->key());
+
+      if (ctrl)
+      {
+            int cx = width() / 2;
+            int cy = height() / 2;
+
+            if (event->key() == Qt::Key_Minus) {
```

Comment:
All four of these keyboard shortcuts should be user-configurable, not hard-coded.

Reply:
How would I do that?